scripts/generate-packages.ps1

```diff
@@ -1,45 +1,66 @@
-(Get-Content $netPs1) `
-  -Replace '-Checksum "not-set"', "-Checksum $netZipHash" `
-  -Replace "__PackageVersion__", "$version" `
-| Set-Content $netPs1
-
-choco pack nuspec\chocolatey\sonarscanner-net-framework.nuspec `
-  --outputdirectory $artifactsFolder `
-  --version $version
-
-choco pack nuspec\chocolatey\sonarscanner-net.nuspec `
-  --outputdirectory $artifactsFolder `
-  --version $version
-
-Write-Host "Update artifacts locations in pom.xml"
-$pomFile = ".\pom.xml"
-(Get-Content $pomFile) `
-  -Replace 'netFrameworkScannerZipPath', "$netFrameworkScannerZipPath" `
-  -Replace 'netScannerZipPath', "$netScannerZipPath" `
-  -Replace 'netScannerGlobalToolPath', "$netScannerGlobalToolPath" `
-  -Replace 'netFrameworkScannerChocoPath', "$artifactsFolder\\sonarscanner-net-framework.$version.nupkg" `
-  -Replace 'netScannerChocoPath', "$artifactsFolder\\sonarscanner-net.$version.nupkg" `
-  -Replace 'sbomPath', "$sbomJsonPath" `
-| Set-Content $pomFile
\ No newline at end of file
+﻿# This script generates the chocolatey packages for the .NET Scanner and the .NET Framework Scanner and updates the pom.xml file with the new artifacts locations.
+
+# The chocolatey packages are generated from the nuspec files located in the nuspec\chocolatey folder and they point to GitHub artifacts that have the following format:
+# Release candidates:
+# - https://github.com/SonarSource/sonar-scanner-msbuild/releases/download/9.0.0-rc.99116/sonar-scanner-9.0.0-rc.99116-net.zip
+# - https://github.com/SonarSource/sonar-scanner-msbuild/releases/download/9.0.0-rc.99116/sonar-scanner-9.0.0-rc.99116-net-framework.zip
+# Normal releases:
+# - https://github.com/SonarSource/sonar-scanner-msbuild/releases/download/8.0.3.99785/sonar-scanner-8.0.3.99785-net.zip
+# - https://github.com/SonarSource/sonar-scanner-msbuild/releases/download/8.0.3.99785/sonar-scanner-8.0.3.99785-net-framework.zip
+
+# In the case of pre-releases, the full version is following the Sem 2 versioning format: 9.0.0-rc.99116.
+# Unfortunatelly Choco only supports Sem 1 versioning format, which does not allow anything except for [0-9A-Za-z-] after the dash in `-rc`.
+# Due to this, when calling `choco pack` the version should not contain the build number (9.0.0-rc).
+# At the same time the the url inside the ps1 file that downloads the scanner should be correct and contain the build number.
+
+param (
+  [string] $sourcesDirectory = $env:BUILD_SOURCESDIRECTORY,
+  [string] $buildId = $env:BUILD_BUILDID
+)
+
+function Update-Choco-Package([string] $scannerZipPath,
+                              [string] $powershellScriptPath,
+                              [string] $nuspecPath) {
+  Write-Host "Generating the chocolatey package from $scannerZipPath"
+
+  $hash = (Get-FileHash $scannerZipPath -Algorithm SHA256).hash
+
+  (Get-Content $powershellScriptPath) `
+    -Replace '-Checksum "not-set"', "-Checksum $hash" `
+    -Replace "__PackageVersion__", "$fullVersion" `
```

Comment:
To see if I understand correctly:

Choco packages are thin wrappers over our github releases, which on the URL, include the build Id.
So even though the package _version_ should NOT include the build Id, because of SemVer1, we still need the full-including-build-id version on the URL, to not fail the choco installations script.
Is that correct?

Reply:
Yes, exactly. The url needs to point to an existing package on GitHub.